File: backend/ProductCatalog.Api/Models/Product.cs
namespace ProductCatalog.Api.Models;

// Puni proizvod - onako kako ga vraća DummyJSON
public class Product
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public string Thumbnail { get; set; } = string.Empty;
    public List<string> Images { get; set; } = new();
    public decimal Rating { get; set; }
    public int Stock { get; set; }
    public string Brand { get; set; } = string.Empty;
}

File: frontend/src/types/product.ts
export interface ProductListItem {
  id: number;
  title: string;
  shortDescription: string;
  price: number;
  thumbnail: string;
}

export interface ProductListResponse {
  items: ProductListItem[];
  total: number;
  page: number;
  pageSize: number;
}

export interface Product {
  id: number;
  title: string;
  description: string;
  category: string;
  price: number;
  thumbnail: string;
  images: string[];
  rating: number;
  stock: number;
  brand: string;
}